### PR TITLE
add optional reason to @Ignored

### PR DESCRIPTION
**Pull request by jschneidereit**
Part of #3029

### CODE REVIEW

File: kotest-framework/kotest-framework-engine/build.gradle.kts [hunk 1]
```diff
@@ -84,8 +84,10 @@ kotlin {
 
       val jvmTest by getting {
          dependencies {
+            implementation(kotlin("stdlib"))
             implementation(project(Projects.Assertions.Core))
             implementation(project(Projects.JunitRunner))
+            implementation(project(Projects.Framework.datatest))
```
Comment (sksamuel) [inline kotest-framework/kotest-framework-engine/build.gradle.kts:47]: Did you want to bring this in for all users

File: kotest-framework/kotest-framework-api/src/commonMain/kotlin/io/kotest/core/annotation/Ignored.kt [hunk 2]
```diff
@@ -6,4 +6,4 @@ package io.kotest.core.annotation
 // @Inherited TODO Not supported by Kotlin yet
 @Target(AnnotationTarget.CLASS)
 @Retention(AnnotationRetention.RUNTIME)
-annotation class Ignored
+annotation class Ignored(val reason: String = "")
```
Comment (sksamuel) [inline kotest-framework/kotest-framework-api/src/commonMain/kotlin/io/kotest/core/annotation/Ignored.kt:9]: Would nullable be a better default?

File: kotest-framework/kotest-framework-engine/build.gradle.kts (hunk 1 above)
Comment (jschneidereit) [reply kotest-framework/kotest-framework-engine/build.gradle.kts:47]: Only on the jvmTest sourceset, that shouldn't affect the published projects, right?

File: kotest-framework/kotest-framework-api/src/commonMain/kotlin/io/kotest/core/annotation/Ignored.kt (hunk 2 above)
Comment (jschneidereit) [reply kotest-framework/kotest-framework-api/src/commonMain/kotlin/io/kotest/core/annotation/Ignored.kt:9]: Yeah I'll make that change 👍

Comment (jschneidereit) [reply kotest-framework/kotest-framework-api/src/commonMain/kotlin/io/kotest/core/annotation/Ignored.kt:9]: Ah, nevermind:

>An annotation parameter cannot be nullable

Comment (sksamuel) [reply kotest-framework/kotest-framework-api/src/commonMain/kotlin/io/kotest/core/annotation/Ignored.kt:9]: Ok well I guess empty string it is :)

File: kotest-framework/kotest-framework-engine/build.gradle.kts (hunk 1 above)
Comment (sksamuel) [reply kotest-framework/kotest-framework-engine/build.gradle.kts:47]: I missed it was test by bad.

**Review by sksamuel — APPROVED** [review]

**Pull request merged**